src/store/DatasetStore.ts

```diff
@@ -78,52 +82,50 @@ export const useDatasetStore = defineStore('dataset', {
     },
     /**
      * Store the result of a datasets fetch operation for an org in store
-     *
-     * @param {string} orgId
-     * @param {object} res
-     * @param {string} sort Sort order used for this res
      */
-    addDatasets(orgId, res, sort) {
+    addDatasets(orgId: string, res: DatasetV2Response, sort: string | null) {
       // reset org data if another sort has been requested
+      if (!this.data[orgId]) this.data[orgId] = []
       if (this.sort !== sort) this.data[orgId] = []
       this.sort = sort
-      this.data[orgId] = [...(this.data[orgId] || []), res]
+      this.data[orgId] = [...this.data[orgId], res]
     },
     /**
      * Get a dataset from the store given its id
-     *
-     * @param {string} datasetId
-     * @returns {object|undefined}
      */
-    get(datasetId) {
+    get(datasetId: string) {
       // flatten pages data for each organization
       // TODO: suboptimal store structure for this use case, see later if org oriented or flat is better
       const flattened = Object.keys(this.data)
         .map((k) => this.data[k].map((a) => a.data).flat())
         .flat()
-      return flattened.find((d) => {
+      const foundDataset = flattened.find((d) => {
         return d.id === datasetId || d.slug === datasetId
       })
+      return foundDataset
     },
     /**
      * Add an "orphan" dataset to the store
-     *
-     * @param {object} dataset
-     * @returns {object}
      */
-    addOrphan(dataset) {
-      this.addDatasets('_orphan', {
-        data: [dataset]
-      })
+    addOrphan(dataset: DatasetV2) {
+      this.addDatasets(
+        '_orphan',
+        {
+          data: [dataset],
+          page: 0,
+          page_size: 0,
+          next_page: null,
+          previous_page: null,
+          total: 0
```

Comment:
```suggestion
          total: 1
```

nitpicks ici, ça ne changera pas grand chose normalement. Faut vraiment refacto ce store de fond en comble :-)